### PR TITLE
feat(medusa): Allow regexp in CORS configuration

### PR DESCRIPTION
**Pull request by adrien2p**
**What**
In some cases, a users that works with a bunch of subdomain would like to be able to configure the cors using a regexp to allow all or a sub set of his sub domains
https://discord.com/channels/876835651130097704/877195433649258536/1002330864216195112

**Info**
Cors middleware impl authorised it:
https://github.com/expressjs/cors/blob/f038e7722838fd83935674aa8c5bf452766741fb/lib/index.js#L19

### CODE REVIEW

File: packages/medusa-plugin-mailchimp/src/api/routes/index.js [hunk 1]
```diff
@@ -3,6 +3,7 @@ import bodyParser from "body-parser"
 import middlewares from "../middleware"
 import { getConfigFile } from "medusa-core-utils"
 import cors from "cors"
+import { parseCorsOrigins } from "@medusajs/medusa/dist/utils"
```
Comment (olivermrbl) [inline packages/medusa-plugin-mailchimp/src/api/routes/index.js:6]: todo: Correct import
```suggestion
import { parseCorsOrigins } from "@medusajs/medusa"
```

Will probably require, that we expose our utils in `medusa/src/index.js`, but I believe that should be fine.

File: packages/medusa-payment-adyen/src/api/routes/hooks/index.js [hunk 2]
```diff
@@ -3,6 +3,7 @@ import cors from "cors"
 import bodyParser from "body-parser"
 import middlewares from "../../middlewares"
 import { getConfigFile } from "medusa-core-utils"
+import { parseCorsOrigins } from "@medusajs/medusa/dist/utils"
```
Comment (olivermrbl) [inline packages/medusa-payment-adyen/src/api/routes/hooks/index.js:6]: todo: Same as other comment

**Review by olivermrbl — CHANGES_REQUESTED** [review]

File: packages/medusa-plugin-mailchimp/src/api/routes/index.js (hunk 1 above)
Comment (olivermrbl) [reply packages/medusa-plugin-mailchimp/src/api/routes/index.js:6]: Alternatively, we could move it to `medusa-core-utils`

**Comment by adrien2p** [comment]
@olivermrbl do you think that pr could be merge at some point?

**Comment by olivermrbl** [comment]
@adrien2p Yes, I will give this a review tomorrow :)

**Comment by olivermrbl** [comment]
I am attempting to configure `ADMIN_CORS` with a regex, but I can't seem to make it work. Am I doing something wrong here?
```
const ADMIN_CORS = `/http\:\/\/localhost\:700\d/gm`
```

This should allow cors for e.g.`http://localhost:7000` and `http://localhost:7001`

**Comment by adrien2p** [comment]
@olivermrbl I ve open a pr in the CORS repo as I can't seam to be able to find the culprit https://github.com/expressjs/cors/issues/292

**Comment by adrien2p** [comment]
It is now working properly, just need to double the \ to avoid them to be escaped

**Comment by adrien2p** [comment]
> That's a great functionality.
> 
> Question: I believe we will increase quite a lot the size of the plugins if they now depend on medusa core. Should we think about separating the "utils" to a new package? ps: considering that some people are not using package managers that reuse installed node_modules

I see what you mean, but plugins already depends on the core as they can extend strategies, interfaces, services, and also are already using some utilities. In that case I am afraid it would mean to separate all of those in different packages. Generally speaking, since the plugins have no dependency to the core, but only as a peer dependency, I don't think it is a problem. they are normally used in the server and the server install medusa core already wdyt?

**Comment by carlos-r-l-rodrigues** [comment]
Sorry, disregard my comment. I see it's listed as peerDependencies. Even npm will reuse medusa-code if that's is installed.

**Comment by adrien2p** [comment]
@olivermrbl finally all green

**Comment by olivermrbl** [comment]
Think we should add a guide for this in our documentation. Wdyt? @adrien2p 

cc. @shahednasser

**Comment by adrien2p** [comment]
Yes I agree, we can take the examples we've put in comments of that pr?

**Comment by shahednasser** [comment]
I'll add an issue for it in Linear 👍🏻

**Comment by srindom** [comment]
@olivermrbl, @adrien2p - can we merge this?

**Comment by olivermrbl** [comment]
Yes, we were just awaiting a green pipeline. After resolving the conflict, we should be good 💪

**Comment by adrien2p** [comment]
@srindom yes sure, i ve tested it and was waiting for someone else to test as well

**Pull request merged**